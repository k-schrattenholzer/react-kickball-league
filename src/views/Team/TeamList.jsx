import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeams } from '../../services/teams.js';

function TeamList() {

  const [teams, setTeams] = useState([]);
  useEffect(() => {
    getTeams().then((resp) => setTeams(resp));
  }, []);

  return (
    <div>
      <h4>top 6 baby teams (nationally ranked)</h4>
      <ul>
        {teams.map((team) => {
          return (
            <li key={team.id}>
              <Link to={`teams/${team.id}`} className='App-link'>
                {team.name}
              </Link>
            </li>
          )
        })

        }
      </ul>
    </div>
  )
}

export default TeamList
