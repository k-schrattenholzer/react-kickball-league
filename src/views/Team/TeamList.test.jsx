import { render } from "@testing-library/react";
import TeamList from './TeamList'

it('should render teamList', () => {
  const { container } = render(<TeamList />);
  expect(container).toMatchSnapshot();
})