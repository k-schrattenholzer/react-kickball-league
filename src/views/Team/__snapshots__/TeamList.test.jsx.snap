// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render teamList 1`] = `
<div>
  <div>
    <h4>
      top 6 baby teams (nationally ranked)
    </h4>
    <ul />
  </div>
</div>
`;
